filesystem/filesystem.go

```diff
@@ -45,12 +46,25 @@ func NewFileSystem(disk disk.Disk) FileSystem {
 	return &fileSystem{root: root, disk: disk}
 }
 
-// CreateDir creates a directory in the nested tree structure,it does not create all parent paths of the final path
+// CreateDir creates a directory in the nested tree structure.
+// Will create the parent directories if they do not already exist.
 func (f *fileSystem) CreateDir(path string) error {
 	structure, err := parseDirStruture(path)
 	if err != nil {
 		return err
 	}
+	fmt.Println("CREATE DIR", structure, len(structure))
+	if len(structure) > 1 {
+		for i := 1; i < len(structure); i++ {
+			if _, exist := f.root.(*directory).contents[structure[i-1]]; !exist {
```

Comment:
suppose we have `/a/b` at startup and we receive a request to create `a/b/c/d` if we parse this and get [a,b,c,d] as directories `a` does exist in ROOT so we do not create `a`, but `b` doesn't exist in ROOT so we attempt to create `/a/b` which will throw an error as it already exists. My point is that checking ROOT level for folder existence might not suffice. let me know if this makes sense and thank you for contributing!

Reply:
I believe that I've found the issue. I need to tidy up some code, and will push an update out shortly.

Reply:
This should be resolved in the latest push that I have made.